### PR TITLE
Guard `$IsWindows`

### LINKED ISSUE #190103: Shell integration with PowerShell 5.1 breaks with `$ErrorActionPreference = 'stop'`
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.81.0
- OS Version: Windows_NT x64 10.0.19045

Steps to Reproduce:

In `$PROFILE.CurrentUserAllHosts`, set the content to 

```
$ErrorActionPreference = 'stop'
Set-StrictMode -Version Latest
```

then start VS Code with shell integration enabled and use Windows PowerShell.

This bug will occur: https://github.com/PowerShell/PowerShellEditorServices/issues/2050

The use of `$IsWindows` at:

https://github.com/microsoft/vscode/blob/8429ea98e9d775b8e6461cd84d2f8691f520dbc2/src/vs/workbench/contrib/terminal/browser/media/shellIntegration.ps1#L123

Needs to be guarded for Windows PowerShell (where it doesn't exist).

/cc @Tyriar

### PR DESCRIPTION
**Pull request by andyleejordan**
So it works with PowerShell 5.1.

Probably resolves https://github.com/microsoft/vscode/issues/190103. I just edited in the browser so haven't tested...

### CODE REVIEW

**Comment by andyleejordan** [comment]
@meganrogge @Tyriar how would I go about checking that the escape sequence is now being emitted correctly?

**Review by syedsadiqali — APPROVED** [review]

**Comment by Tyriar** [comment]
@andyleejordan set log level to trace then open Output > Terminal. You should see this when you create a terminal:

image

File: src/vs/workbench/contrib/terminal/browser/media/shellIntegration.ps1 [hunk 1]
```diff
@@ -120,7 +120,11 @@ if (Get-Module -Name PSReadLine) {
 }
 
 # Set IsWindows property
-[Console]::Write("$([char]0x1b)]633;P;IsWindows=$($IsWindows)`a")
+if ($PSVersionTable.PSVersion -lt "6.0") {
+	[Console]::Write("$([char]0x1b)]633;P;IsWindows=$true`a")
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/media/shellIntegration.ps1:124]:
```suggestion
	# Windows PowerShell is only available on Windows
	[Console]::Write("$([char]0x1b)]633;P;IsWindows=$true`a")
```

**Review by Tyriar — CHANGES_REQUESTED** [review]

**Review by Tyriar — APPROVED** [review]
I confirmed it works 👍

**Review by alexdima — APPROVED** [review]

**Pull request merged**